java-checks/src/test/files/checks/RedundantTypeCastCheck.java

```diff
@@ -156,4 +156,14 @@ public static Number choose(Integer i, Float f, boolean takeFirst) {
       return takeFirst ? (Number) i : f;
     }
   }
+
+  class I {
```

Comment:
Please also add the (reduced) cases causing initial FP into the test of `RedundantTypeCast`